### DISCUSSION TITLE
`values` is not generated for enums with open case

### DISCUSSION THREAD

**Discussion by goshacodes** [scala/scala3#21955] [Feature Requests]
## Compiler version
3.3.4

## Problem
Hi!

A common pattern when creating enums is to define `value` field and leave one case open.
 
When defining such patterns you still need basic `values`. This is one of greatest features of `enumeratum` and it is missing in stdlib.

I think there was a reason why this was done like this, but it would be great to somehow support it out of the box.

Other thing to improve is to make `valueOf` bounded to this custom `value` for basic cases.

Is this possible?

## Minimized code
```scala

enum Segment(val value: String):
  case Literal(override val value: String) extends Segment(value)
  case One extends Segment("*")

object Segment:
  def valueOf(value: String): Segment =
    Segment.values.find(_.value == value).getOrElse(Literal(value))
```

## Output

```scala
[error] 14 |    Segment.values.find(_.value == value).getOrElse(Literal(value))
[error]    |    ^^^^^^^^^^^^^^
[error]    |value values is not a member of object ru.tinkoff.sso.commons.authorization.model.Segment - did you mean Segment.valueOf?
[error]    |Although class Segment is an enum, it has non-singleton cases,
[error]    |meaning a values array is not defined
```

## Expectation
```scala
//compiles and contains basic cases
Segment.values == Array(Segment.One)

Segment.valueOf("*") == Segment.One
```

**Comment by bishabosha** [comment]
The original reasoning was that values is exhaustive rather than "whatever values are defined" I'm sure we can make it opt in, but there are also I guess library solutions to this if we don't make it configurable
